source/Server.Contracts/Accounts/AccountDetails.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using Sashimi.Server.Contracts.Variables;
+
+namespace Sashimi.Server.Contracts.Accounts
+{
+    public abstract class AccountDetails
+    {
+        [JsonIgnore]
+        public abstract AccountType AccountType { get; }
+
+        public virtual IEnumerable<Variable> ExpandVariable(Variable variable)
+        {
+            return Enumerable.Empty<Variable>();
+        }
+
+        public virtual IEnumerable<(string key, string template)> ContributeResourceLinks()
+        {
+            return Enumerable.Empty<(string key, string template)>();
+        }
+
+        public abstract IEnumerable<Variable> ContributeVariables();
```

Comment:
I made this abstract, I think `Account` on the server side should contribute the `AccountType` variable

Reply:
This used to be
```
public virtual IEnumerable<Variable> ContributeVariables()
{
    yield return new Variable("Octopus.Account.AccountType", AccountType.ToString());
}
```
With this shape we are putting that responsibility on all account implementers, I am not sure that is their responsibility?

Reply:
actually all good, that was wrong that contribution is already done on the account 👍

Reply:
Amazon now needs an empty implementation, which I think is OK. This saves Azure from having to do:
```
                foreach (var variable in base.ContributeVariables())
                {
                    yield return variable;
                }
```